doc/error_reporting.md

```diff
@@ -0,0 +1,90 @@
+# Error Reporting in AutoYaST
+
+This document tries to summarize the error reporting mechanisms that, as a developer, you can use
+when writing code for AutoYaST. Bear in mind that both mechanisms have different purposes. On the
+one hand, the `Yast::Report` module is the way to go when you want to notify a general problem or
+ask a Yes/No question. On the other hand, the purpose of the AutoYaST issues mechanism is to notify
+semantic issues in the profile.
+
+## Yast::Report Module
+
+The `Yast::Report` module offers a simple and configurable mechanism for error reporting. It relies
+on the `Yast::Popup` mechanism but offers a few additional features:
+
+* Show/Hide messages depending on its severity (error, messages, yes/no questions, etc.).
+* Messages logging.
+* Time-outs.
+
+```ruby
+Yast.import "Report"
+Yast::Report.Error("Something was wrong")
+```
+
+Usually, the reporting settings are defined in the AutoYaST profile. For instance, with the
+following settings, AutoYaST stops only when an error message is reported. For the rest, it applies
+a 10 seconds time-out. Additionally, all messages are logged.
```

Comment:
I would mention here that Report is more general solution that works well also in CLI and non AY scenario.

Reply:
and that for such cases Report uses proper output.